### PR TITLE
Registry of Sunpy Affiliated Packages

### PR DESCRIPTION
**Pull request by prateekiiest**
I have uploaded some of the affiliated packages in the given format.
Added some padding too. It wasn't separate from the footer.

### CODE REVIEW

**Comment by prateekiiest** [comment]
Looks like this.
screenshot 2017-04-27 17 06 42

**Comment by prateekiiest** [comment]
Reviews pls. If its ok, I will then upload the remaining of the package names.

**Comment by ayshih** [comment]
Er, I may have missed some discussion, but I'm concerned that you're including packages that aren't actually affiliated packages (e.g., `heroes`).  It's not enough that a package uses SunPy, or even lives under SunPy on GitHub, but it has to meet (or be actively working to meet) the requirements in the SEP *and* be accepted by the board.

**Comment by prateekiiest** [comment]
Oh my mistake. I thought HEROES was an affiliated package.
@ayshih , I went through this SEP https://github.com/sunpy/sunpy-SEP/blob/master/SEP-0004.md which also states the same as said by you.
But I thought these were already accepted.

For new packages, will it be ok to create a database where new users can put in their package name and the board sees it. Or just simple mailing?

**Comment by Cadair** [comment]
@prateekiiest the process is just to contact the mailing list and ask, we are not expecting a high volume of packages!!

**Comment by Cadair** [comment]
Also put me down as maintainer for ginga-sunpy.

**Comment by Cadair** [comment]
Also, also, I would put a short paragraph describing affiliated packages from the SEP in this section.

**Comment by prateekiiest** [comment]
So we should describe why we are needing affiliated packages and also the requirements to be fulfilled in order to be accepted?

**Comment by Cadair** [comment]
Describe what they are, and what the requirements are. Make it pretty high level though, it's mainly for users not developers.

Affiliated packages are not a concept most solar physicists are used to (they are more used to one monolithic package) so we need to make it clear to them on first read.

File: contribute.html [hunk 1]
```diff
@@ -80,5 +84,28 @@ <h1 id="packages">Affiliated Packages</h1>
...
+
+	
+	
+	<th class="head">Code Repository</th>
+	<th class="head">Maintainer</th>
+	</tr>
+	</thead>
+	<tbody valign="top">
+	
+	<tr></tr><tr><td>sunkit-sst</td><td>Yes</td><td><a href="https://github.com/sunpy/sunkit-sst">sunkit-sst</a></td><td><a href="https://github.com/nabobalis">Nabil Freij</a></td></tr><tr></tr><tr><td>SolarExtrapolation</td><td>Yes</td><td><a href="https://github.com/sunpy/solarbextrapolation">solarbextrapolation</a></td><td><a href="https://github.com/Cadair">Stuart Mumford</a></td></tr><tr><td>ginga-sunpy</td><td>Yes</td><td><a href="https://github.com/sunpy/ginga-sunpy">ginga-sunpy</a></td><td><a href="https://github.com/Cadair">Stuart Mumford</a></td></tr><tr><td>heroes</td><td>Yes</td><td><a href="https://github.com/sunpy/heroes">Heroes(High Energy Replicated Optics to Explore the Sun)</a></td><td><a href="https://github.com/ehsteve"> Steven Christie</a></td></tr><tr></tr><tr></tr><tr></tr>
```
Comment (Cadair) [inline contribute.html]: Please add line breaks and indentation to this!!

Comment (prateekiiest) [inline contribute.html]: ok sure

**Comment by nabobalis** [comment]
I would prefer the package list be in alphabetical order.
Also, what does centered columns look like?

**Comment by prateekiiest** [comment]
@nabobalis I am adding the alignment for in the middle.
But I feel it will be really tedious to sort the names every time a new package comes.

**Comment by nabobalis** [comment]
I am not sure it will be that much of a problem, I very much doubt there would be a new package that often.

**Comment by prateekiiest** [comment]
Yeah, but its like adding a new number to a sorted list and then sorting it again.
Kind of an algorithm problem ;P

**Comment by prateekiiest** [comment]
Done @Cadair @nabobalis

**Comment by prateekiiest** [comment]
Looks like this now
screenshot 2017-04-28 16 26 05

**Comment by nabobalis** [comment]
Looks good, but I don't understand why the first part is italicized and satisfied should be lower case.

**Comment by prateekiiest** [comment]
Its just for style. I felt it was too monotonous.
Should I change it then?

**Comment by nabobalis** [comment]
I don't like it. Maybe @Cadair has a different opinion.

**Comment by prateekiiest** [comment]
Ok I am changing it then

**Comment by Cadair** [comment]
SunPy should have a capital 'P' everywhere.

**Comment by nabobalis** [comment]
Wouldn't a camelcase line have a capital T and B?

**Comment by prateekiiest** [comment]
I am not sure whether to use CameCase for Articles and Prepositions?
Anyways, I changed it to lowercase.
Its better I guess

**Comment by nabobalis** [comment]
I'm not aware that camelcase is used in natural language that often expect for specific names or made up words.

**Comment by prateekiiest** [comment]
@nabobalis is there any package I am missing as for now?

**Comment by nabobalis** [comment]
I think thats the correct ones, I'm not aware of others. 

The white space usage in the HTMl file is inconsistent.

File: contribute.html [hunk 2]
```diff
@@ -71,14 +71,106 @@ <h1 id="proposals">SunPy Enhancement Proposals (SEP)</h1>
...
+		<td>Yes</td>
+		<td><a href="https://github.com/sunpy/irispy">Irispy</a></td>
+		<td><a href="https://github.com/DanRyanIrish">DanRyanIrish</a></td>
+		</tr>
+		
+	
+	
+	<!-- SolarExtrapolation =========== -->
+		<tr>
+		<td>SolarExtrapolation</td>
```
Comment (ayshih) [inline contribute.html]: Shouldn't the name of this package include the letter "B" between "Solar" and "Extrapolation"?

File: contribute.html [hunk 3]
```diff
@@ -71,14 +71,106 @@ <h1 id="proposals">SunPy Enhancement Proposals (SEP)</h1>
...
+		<tr>
+		<td>ginga-sunpy</td>
+		<td>Yes</td>
+		<td><a href="https://github.com/sunpy/ginga-sunpy">ginga-sunpy</a></td>
+		<td><a href="https://github.com/Cadair">Stuart Mumford</a></td>
+		</tr>
+		
+	<!-- Irispy =========== -->
+		<tr>
+		<td>irispy</td>
```
Comment (Cadair) [inline contribute.html]: should be 'IRISPy'

File: contribute.html [hunk 4]
```diff
@@ -71,14 +71,106 @@ <h1 id="proposals">SunPy Enhancement Proposals (SEP)</h1>
...
+		<td>Yes</td>
+		<td><a href="https://github.com/sunpy/irispy">Irispy</a></td>
```
Comment (Cadair) [inline contribute.html]: and here.

File: contribute.html [hunk 5]
```diff
@@ -71,14 +71,106 @@ <h1 id="proposals">SunPy Enhancement Proposals (SEP)</h1>
...
+		<td><a href="https://github.com/DanRyanIrish">DanRyanIrish</a></td>
```
Comment (Cadair) [inline contribute.html]: Daniel Ryan

File: contribute.html [hunk 6]
```diff
@@ -71,14 +71,106 @@ <h1 id="proposals">SunPy Enhancement Proposals (SEP)</h1>
...
+	
+	<th class="head">Code Repository</th>
+	<th class="head">Maintainer</th>
+	</tr>
+	</thead>
+	<tbody valign="top">
+	
+		
+			
+	<!-- ginga-sunpy =========== -->
```
Comment (Cadair) [inline contribute.html]: @ehsteve @ayshih Do we want this in here, it's not really maintained. We should probably drop it?

Comment (ayshih) [inline contribute.html]: My preference would be to leave it out.

**Comment by ayshih** [comment]
As a more top-level comment, I feel that it makes more sense for this registry to link people to documentation trees rather than code repositories.  (Note that it's a requirement that any affiliated package have competent documentation, so this should always be possible!)  I anticipate that someone finding out about an affiliated package through this page would be next interested in learning more about that package, so directing them to a GitHub repo is not particularly useful or relevant.

**Comment by Cadair** [comment]
Agreed, links to documentation would be much better. All the affiliated packages can have a docs url that looks like: http://docs.sunpy.org/projects/solarbextrapolation/

**Comment by nabobalis** [comment]
Maybe not all.

**Comment by Cadair** [comment]
@nabobalis we *can* add any project on RTD as a subproject and therefore give them that URL.  At the moment the only ones that do are irispy, solarb and sunkit-sst

**Comment by prateekiiest** [comment]
Currently are there any doc URLs? Or any website for the packages?

**Comment by nabobalis** [comment]
@Cadair , I too can _italic_ my words.

**Comment by nabobalis** [comment]
But once I merge in the PR, docs will build for it anyway. So it isn't an issue.

**Comment by prateekiiest** [comment]
Should then remove ginga sunpy?

File: contribute.html (hunk 2 above)
Comment (prateekiiest) [inline contribute.html]: Added

**Comment by DuyguKeskek** [comment]
Hi @prateekiiest @Cadair, I just want to give some recommendations. First, it looks like a _very_ old fashioned table that is full of black borders and hard to understand. It would be better if you use [bootstrap tables](https://v4-alpha.getbootstrap.com/content/tables/) for a clean view. Since it is mainly for users, it should also take the attention of them and we can do it by adding some featured content, [images](https://drive.google.com/drive/folders/0B_v8Zis3SvXZeFBUU0FHOVhsU00?usp=sharing), [thumbnails](http://getbootstrap.com/components/#thumbnails), or [cards](https://material.io/guidelines/components/cards.html#) etc. I also like how [AstroPy](http://www.astropy.org/affiliated/index.html) implemented the affiliated packages and we can take it as a reference. Additionally, it is way better to give some introductory info about the packages since 90% of the users will not even click on them for further info.

**Comment by prateekiiest** [comment]
Hi @DuyguKeskek 
Thanks for your feedback . Yeah, I know the tables are not looking good currently and needs style and bootstrap tables is a great idea. But there are  many insertions and deletions going on now. So once the contents of the table get finalised, I will surely concentrate on its design.
Yes some intro info is required for the packages, but as discussed above it will better if we could link to their corresponding docs URLs. Since its basically a website, I think it will better if we make the info crisp and small and the URLs will be sufficient for that. :)

**Comment by prateekiiest** [comment]
Astropy has some 2 or 3 featured package description and then the table.
I feel a separate page will be better then, @Cadair ?

**Comment by prateekiiest** [comment]
Looks good now?
screenshot 2017-04-29 07 23 19

**Comment by Cadair** [comment]
@prateekiiest I would drop the number column. Also a one sentence description would be good, people might not know what the package does from the name.

**Comment by prateekiiest** [comment]
Ok, sure
And should I upload the docs url instead of the code repo url?

**Comment by Cadair** [comment]
You could have the doc link of the name from the package and keep the repo?

**Comment by prateekiiest** [comment]
does irispy have any docs url?
I didn't find one.

**Comment by Cadair** [comment]
http://docs.sunpy.org/projects/irispy/en/latest/

**Comment by prateekiiest** [comment]
@Cadair , looks like this.

screenshot 2017-04-30 18 44 44

**Comment by prateekiiest** [comment]
Sorry, the name should be extrapolation, instead of exploration :)
I already corrected that in the code

**Comment by prateekiiest** [comment]
Does this need any more updating?

**Comment by nabobalis** [comment]
The main thing thats needs changing is the way the text breaks on the screenshot. I do not like that for SolarBExploration, the 'n' is on a new line. 

Is that due to your current screen resolution?

**Comment by prateekiiest** [comment]
Yeah, that's just due to current screen resolution.
Also for a phone or tablet, the resolution will be smaller, and thus  the names bound to get broken, since the table is responsive

**Comment by nabobalis** [comment]
Can you please show a screenshot of what the table looks like with your browser window made smaller? 

I am currently not in a position to compile the website right now.

**Comment by prateekiiest** [comment]
Sorry, @nabobalis , I fear its still breaking.
I guess reducing the font size will clear this issue, but that will reduce the font size of other sections in About, Blog as well.
Will that do?

**Comment by nabobalis** [comment]
Hmm, could you remove the stable column for me? 

I think the description of the SST one could be shrunk to: 

Allows the reading of spectral files that are used by the Swedish Solar Telescope.

**Comment by prateekiiest** [comment]
screenshot 2017-05-19 13 55 54

**Comment by prateekiiest** [comment]
ok sure, I am on it

**Comment by nabobalis** [comment]
Also, I think the package name should redirect to the code repo, so that column can go as well. 

Then Docs Url should just be Documentation

**Comment by prateekiiest** [comment]
screenshot 2017-05-19 14 02 12
Looks good now, I guess

**Comment by nabobalis** [comment]
One more change, for maintainer for SolarB... Can you replace it to Alex Hamilton, https://github.com/Alex-Ian-Hamilton

**Comment by Cadair** [comment]
The description for IRISpy is wrong.

**Comment by Cadair** [comment]
Should be "A package for handling data from the IRIS satellite"

**Comment by nabobalis** [comment]
Good spot. #WOOPS

**Comment by nabobalis** [comment]
TO THE EDIT MACHINE

**Pull request merged**